### DISCUSSION TITLE
Red Hat Enterprise Linux 8.x not in list installable operating systems?

### DISCUSSION THREAD

**Discussion by Verhoeckx** [cockpit-project/cockpit-machines#268] [Q&A] [answered]
Hello Cockpit developers,

I see that Red Hat Enterprise Linux 8.x is not in the list of installable operating systems in the window Create new Virtual Machine?
Why not? Shouldn't it be there? It's the main operating system of Red Hat!

I guess it's not a problem for me because I already downloaded Red Hat Enterprise Linux 8.4 on my local machine. I just have to copy it to the server.

**Comment by garrett** [comment]
The list is from libosinfo, which is used by other virtual machine managers, such as Virt-Manager and GNOME Boxes.

Which Linux distribution are you running? Am I correct that you're using Cockpit on RHEL 8? Perhaps libosinfo is out of date? Is RHEL 7.x listed?

If you change the installation type from download to another method, Red Hat Enterprise Linux is there (at least here).

![Screenshot from 2021-07-22 17-53-42](https://user-images.githubusercontent.com/10246/126669879-730d3cb3-4994-404c-8b7c-e8d20fb697ed.png)

To download Red Hat Enterprise Linux, you need to have a developer account or acquire it through your company subscription, so it's not quite as simple as using the download method, if that's what you're trying.

Perhaps we should still show it in the list under downloads too, but have a note with a link (that would go offsite) on where to get it.

**Reply by garrett** [reply]
Thanks for reporting this! I've opened an issue about Red Hat Enterprise Linux not showing up in the download list @ https://github.com/cockpit-project/cockpit-machines/issues/269, as this is a UX issue — it's not obvious enough.

If Red Hat Enterprise Linux isn't showing up in the list for other installation types, please also let us know, as that's either a bug or an out of date libosinfo on your machine (and it would be good to know).

**Reply by Verhoeckx** [reply]
I'm wondering:
What happens when you download RHEL and start using it _without_ a subscription? Will you get a notice or something? Will it stop working after some time (maybe a little bit dramatic)?

If the use of RHEL is somehow blocked after a while you could added it to the list of downloadable operating systems. Next to it you could add `Needs a subscription` (or something like that). But of course I'm not sure if this is legal. Having said that: RHEL can be downloaded with GNOME boxes so I think it is allowed?

---

You are right: I do use RHEL (8.4) and I use the No-Cost Red Hat Developer Subscription. This subscription allows me to install RHEL on 16 (virtual) machines.

I solved my problem by coping (scp) the RHEL ISO image from my laptop to my server (an Intel NUC).

When I choose `Download an OS` all kinds of distributions are listed but none of Red Hat itself (so no RHEL 7 or RHEL 8).
When I choose `URL` I get all the distributions including all versions of RHEL ( RHEL 6 until RHEL 9).

**Reply by KKoukiou** [reply]
Ok it's clear that you mean that the OS is missing after you choose 'Download an OS'. The reason is that osinfo DB does not provide the tree media in their database, see my comment https://github.com/cockpit-project/cockpit-machines/issues/269

I don't know if there is a reason for that, I mentioned a developer from that project in that PR so we can find out.

So let's close this discussion, and wait for the answer in the issue @garrett opened. Thanks :)

**Comment by KKoukiou** [comment, marked as answer]
@Verhoeckx you can find the answer here why the RHEL is missing from that list.
#269 (comment)

**Reply by Verhoeckx** [reply]
Hi @KKoukiou,

I don't not understand what `tree media` is, but I do understand that at the moment it's not technical possible to provide RHEL as an option in the list.

Having said that: I wonder how GNOME Boxes is doing this because with GNOME Boxes it is possible to download RHEL.

From a marketing perspective I think it's not a good idea to not show RHEL in the list: your are missing your own product! And what if I, as a user, want to install RHEL in a virtual machine?

It's not a problem for me because I already downloaded RHEL and copied it to my server. I just wanted to notify Red Hat / Cockpit about this (to help).

**Reply by KKoukiou** [reply]
tree media is a directory based installation tree like: https://download.fedoraproject.org/pub/fedora/linux/releases/33/Server/x86_64/os


@fabiano do you happen to know how gnome boxes implements installing RHEL without that the user provides a URL/iso? If not I will take a look personally, I just know you 're very close to that project.
